### PR TITLE
CP: #11322 #11338 #11342

### PR DESCRIPTION
**Pull request by QilongTang**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

CP: #11322 #11338 #11342

Validated
image

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### FYIs

@DynamoDS/dynamo

### CODE REVIEW

**Review by mmisol — CHANGES_REQUESTED** [review]
I think we should make sure this get tested before we merge it in 2.10. Specially since it doesn't contain unit tests

**Comment by QilongTang** [comment]
@mmisol Yes, just create the PR to get self CI running for now. Behavior is getting tested

**Comment by QilongTang** [comment]
Validated by me and @mjkkirschner. Merging and asking @saintentropy to append unit test to master branch for these. This is to unblock the RC2.10.0 release and localization.

**Comment by mmisol** [comment]
If this was tested manually that sounds good. No need to ask for unit tests retroactively.

File: src/Libraries/CoreNodes/Math.cs
Comment (mmisol) [inline src/Libraries/CoreNodes/Math.cs:20]: Actually, `[` was right here because according to the docs the value can be 0: https://docs.microsoft.com/en-us/dotnet/api/system.random.nextdouble?view=net-5.0

File: src/Libraries/CoreNodes/Math.cs
Comment (mmisol) [inline src/Libraries/CoreNodes/Math.cs:35]: This should be `Lower/Higher end of the range for the random number.`

File: src/Libraries/CoreNodes/Math.cs
Comment (QilongTang) [reply src/Libraries/CoreNodes/Math.cs:20]: @mmisol We did see this doc but from our own testing with millions of numbers the 0 was never hit.

Comment (mmisol) [reply src/Libraries/CoreNodes/Math.cs:20]: @QilongTang The value space is huge. I can say for sure you didn't get all possible values 😄

Comment (QilongTang) [reply src/Libraries/CoreNodes/Math.cs:20]: After numerous testing, @mjkkirschner and I decided to update the range here as majority of our users would benefit from this..

Comment (mmisol) [reply src/Libraries/CoreNodes/Math.cs:20]: I don't see how giving misleading documentation could benefit users.

Comment (mjkkirschner) [reply src/Libraries/CoreNodes/Math.cs:20]: @mmisol you could be right of course - I am not a statistician -  my hypothesis was that we should see a 0 once or twice in a million results - but... I guess I am super wrong:

https://stackoverflow.com/questions/58241868/can-random-uniform0-1-ever-generate-0-or-1

the response here is a few quadrillion... I'm not so sure about that either...

I think the result of system.random and our own method is confused by the fact that we do some scaling - so it's not clear if we actually have returned 0 and the result is loosing precision.

**Pull request merged**
